src/lib/Mapper/SearchHitToContentSuggestionMapper.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Search\Mapper;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Search\SearchHit;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Ibexa\Contracts\Search\Mapper\SearchHitToContentSuggestionMapperInterface;
+use Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion;
+use Ibexa\Contracts\Search\Provider\ParentLocationProviderInterface;
+
+final class SearchHitToContentSuggestionMapper implements SearchHitToContentSuggestionMapperInterface
+{
+    private ConfigResolverInterface $configResolver;
+
+    private ParentLocationProviderInterface $parentLocationProvider;
+
+    public function __construct(
+        ParentLocationProviderInterface $parentLocationProvider,
+        ConfigResolverInterface $configResolver
+    ) {
+        $this->configResolver = $configResolver;
+        $this->parentLocationProvider = $parentLocationProvider;
+    }
+
+    public function map(SearchHit $searchHit): ?ContentSuggestion
+    {
+        $content = $searchHit->valueObject;
+
+        if (!$content instanceof Content) {
+            return null;
+        }
+
+        $rootLocationId = $this->configResolver->getParameter('content.tree_root.location_id');
+
+        $mainLocation = $content->getVersionInfo()->getContentInfo()->getMainLocation();
+
+        if ($mainLocation === null) {
+            return null;
+        }
+
+        $parentsLocation = array_map('intval', $mainLocation->path);
+        $position = array_search($rootLocationId, $parentsLocation, true);
+        if ($position !== false) {
+            $parentsLocation = array_slice($parentsLocation, (int)$position);
+        }
+
+        $parentCollection = $this->parentLocationProvider->provide($parentsLocation);
+
+        return new ContentSuggestion(
+            $searchHit->score ?? 50.0,
```

Comment:
Any specific reason why 50.0?
Also, score doc type says that this is never null, so it either needs to be fixed to never have null if for some reasons this check is needed or we just need doc correction for time being,

Reply:
The legacy search lacks a scoring feature. I've introduced the capability to integrate multiple data sources for autocomplete, that can operate at the same time one after another. This introduces a "score" metric, allowing for the customization of result sequencing which at the end of service is cut to a given limit.

Reply:
Ok, so I guess we need to fix doc on searchHit::score, as it stays there that this cannot be null, but it is for legacy search? 
@alongosz